server/src/routes/routes.ts

```diff
@@ -86,4 +87,102 @@ router.post('/auth/login', async (req: Request, res: Response) => {
   res.json({ found });
 });
 
-export { router };
\ No newline at end of file
+
+router.post('/api/saveFile', (req : Request, res: Response) => {
+  const { username, workspace, filename, fileData } = req.body;
+  const path = './user-files/' + username + '/workspaces/' + workspace + '/' + filename;
+
+  try {
+    writeFileSync(path, fileData);
+    // save to volume here
+    res.json({ path: path });
+  } catch (err) {
+    res.json({ error: err });
```

Comment:
You should respond with a different status code if there was an error. We could have more comprehensive error checking in the future but a status code of 500 would probably be suitable for now to indicate a server error.
```js
res.status(500).json({ error: err })
```